tests/basic/storage_pool_test.go

```diff
@@ -9761,3 +9761,192 @@ var _ = Describe("{AddDriveMetadataPool}", func() {
 		AfterEachTest(contexts)
 	})
 })
+
+var _ = Describe("{AddDriveBeyondMaxSupported}", func() {
+
+	/*
+		Add Drive using legacy add drive feature
+		test max drive per pool
+		test max pool per node
+		test max drives per node
+	*/
+	var testrailID = 19170501
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/tests/view/19170501
+	var runID int
+	JustBeforeEach(func() {
+		StartTorpedoTest("AddDriveBeyondMaxSupported", "Initiate pool Cloud add-drive beyond permitted and expect error", nil, testrailID)
+		runID = testrailuttils.AddRunsToMilestone(testrailID)
+	})
+	var contexts []*scheduler.Context
+
+	stepLog := "Test max limit of pool,node and max pool in a cluster"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("pooladddrivebeyondmax-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+		stNodes := node.GetStorageNodes()
+		dash.VerifyFatal(len(stNodes) > 0, true, "Storage nodes found?")
+		selectedNode := stNodes[0]
+		if selectedNode.Id == "" {
+			err := errors.New("failed to get node id")
+			log.FailOnError(err, "failed to get node id")
+		}
+
+		Step("Test beyond max drives per pool", func() {
+			var selectedPool *api.StoragePool
+			var maxDrivesAllowed int
+			selectedPool, err := GetPoolWithIOsInGivenNode(selectedNode, contexts)
+			log.FailOnError(err, "error in getting pool with IO with error")
+			if selectedPool != nil {
+				drvMap, err := Inst().V.GetPoolDrives(&selectedNode)
+				if len(drvMap) == 0 {
+					log.FailOnError(fmt.Errorf("error getting the drive map from the node"), "error getting pool drivemap from node [%s]", selectedNode.Name)
+				}
+				log.FailOnError(err, "error getting pool drives from node [%s]", selectedNode.Name)
+				drvs := drvMap[fmt.Sprintf("%d", selectedPool.ID)]
+				maxDrivesAllowed = POOL_MAX_CLOUD_DRIVES - len(drvs)
+			} else {
+				log.FailOnError(err, "error getting pool from node [%s]", selectedNode.Name)
+			}
+			for i := 1; i <= maxDrivesAllowed+1; i++ {
+				drvSize, err := getPoolDiskSize(selectedPool)
+				log.FailOnError(err, "fialed while getting pool size")
+				driveSize := drvSize * uint64(i)
+				expectedSize := (selectedPool.TotalSize / units.GiB) + driveSize
+				err = Inst().V.ExpandPool(selectedPool.Uuid, api.SdkStoragePool_RESIZE_TYPE_ADD_DISK, expectedSize, false)
+				log.FailOnError(err, "error while expanding pool")
+				isjournal, err := isJournalEnabled()
+				log.FailOnError(err, "Failed to check is journal enabled")
+				resizeErr := waitForPoolToBeResized(expectedSize, selectedPool.Uuid, isjournal)
+				if i == maxDrivesAllowed+1 {
```

Comment:
rather than putting if condition first add the max drives then try to add extra disk

Reply:
we have to bring all the parmaters down again to the last iteration.So will discuss with you on this.No changes made

Reply:
will discuss with you @lsrinivas-pure .Resolving for now